bot/cogs/levelling.py

```diff
@@ -1,22 +1,25 @@
-import logging
+from __future__ import annotations
 
-import discord.ext.commands as commands
-from discord import DMChannel
+from fred_core_imports import *
+from libraries import common
 
-import config
+import nextcord.ext.commands as commands
+from nextcord import DMChannel
 from datetime import *
-import math
-
-from libraries import helper
 
 
 class UserProfile:
     def __init__(self, user_id, guild, bot):
         self.guild = guild
         self.bot = bot
         self.user_id = user_id
-        self.member = guild.get_member(self.user_id)
-        if DB_user := config.Users.fetch(self.user_id):
+
+        assert bot.intents.members, "The bot has no member read permissions!!"
```

Comment:
Not actually permissions, it's intents. I'm nitpicking but there's a very low chance it could cause some hellish debugging session so please clarify